client/image_prune.go

```diff
@@ -22,8 +23,13 @@ func (cli *Client) ImagesPrune(ctx context.Context, pruneFilters filters.Args) (
 		return report, err
 	}
 
-	serverResp, err := cli.post(ctx, "/images/prune", query, nil, nil)
-	if err != nil {
+	cCtx, cancel := context.WithCancel(ctx)
+	RegisterSignalHandler(func() {
+		cancel()
+	})
```

Comment:
Without trapping the signal, the client just gets killed and I want it to still display information about the part it has managed to go through before the cancellation (this part isn't working yet, though)